plugin/vifm.vim

```diff
@@ -255,6 +255,74 @@ function s:DropGoneBuffers(buffer_snapshot) abort
 	endif
 endfunction
 
+" Opens files after exiting Vifm.  Returns a dict of opened buffer names.
+function! s:OpenFiles(editcmd, flist, opentype) abort
+	let opened_bufnrs = {}
+
+	" User exits vifm without selecting a file.
+	if empty(a:flist)
+		echohl WarningMsg | echo 'No file selected' | echohl None
+		return opened_bufnrs
+	endif
+
+	let flist = a:flist
+	call map(flist, 'resolve(fnamemodify(v:val, ":."))')
```

Comment:
I changed some of the logic for handling escaping and modifying file names in this function.

Previously, `resolve()` and `fnamemodify()` were called after `fnameescape()`. These functions don't actually work with escaped filenames.

The `resolve()` function wasn't actually called in one case, and there was special logic for resolving filenames in one case. However, I discovered this was still broken in my tests - this logic broke with filenames that needed to be escaped or linked files (it only worked with files in linked folders that didn't need to be escaped).

Now, `resolve()` and `fnameescape()` are used properly on all files, and `fnameescape()` is now called when it's needed.

Reply:
Thank you for cleaning up that mess!  I'm not sure why I got the impression that Vim was resolving only part of the path there.